### PR TITLE
Unit test fs, ftp, network functions

### PR DESCRIPTION
**Pull request by newfla**
Close the easy part of #31

### CODE REVIEW

File: packages/app/src-tauri/src/commands/ftp.rs [hunk 1]
```diff
@@ -31,7 +31,13 @@ pub async fn ftp_upload(payload: Payload) -> Result<String, String> {
         return Err("Failed to login to server: {}".to_string());
     }
 
-    ftp_stream.set_mode(Mode::ExtendedPassive);
+    // https://www.iana.org/assignments/ftp-commands-extensions/ftp-commands-extensions.xhtml
+    if ftp_stream
+        .feat()
+        .is_ok_and(|opts| opts.get("EPSV").is_some())
+    {
+        ftp_stream.set_mode(Mode::ExtendedPassive);
```
Comment (brainrepo) [inline packages/app/src-tauri/src/commands/ftp.rs:39]: @newfla what do you think if we enable it by a configuration flag. 
We cannot assume that the connection will always be a passive connection?
I mean to keep this check but also verify if a passiveMode plag is true in the payload.

File: packages/app/src-tauri/src/commands/ftp.rs [hunk 2]
```diff
@@ -77,3 +84,156 @@ pub async fn ftp_upload(payload: Payload) -> Result<String, String> {
...
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn test_ftp_upload_ok() {
+        let port = 2121;
+        let payload = Payload {
+            host: "localhost".to_string(),
+            port: port,
+            user: "ForuviereTestUser".to_string(),
+            password: "StealThisUselessPassword".to_string(),
```
Comment (brainrepo) [inline packages/app/src-tauri/src/commands/ftp.rs:141]: :D

**Review by brainrepo — APPROVED** [review]
Awesome work 🤟🏼! 
This test coverage improvement will be very helpful! 
I left just two nits, but I am happy
to merge it now

File: packages/app/src-tauri/src/commands/ftp.rs (hunk 1 above)
Comment (newfla) [reply packages/app/src-tauri/src/commands/ftp.rs:42]: As default we can set to Mode::Passive, then check for the EPSV flag. Set_mode simply stores the flag so should not be a performance/protocol issue. We can ignore active mode

File: packages/app/src-tauri/src/commands/mod.rs [hunk 3]
```diff
@@ -2,3 +2,14 @@ pub mod fs;
 pub mod ftp;
 pub mod network;
 pub mod s3;
+
+#[macro_export]
+macro_rules! test_file {
```
[the remark this answers is not in the document]
Comment (newfla) [reply packages/app/src-tauri/src/commands/mod.rs]: Moved to src/utils/test.rs

**Comment by brainrepo** [comment]
Simply perfect!

**Review by brainrepo — APPROVED** [review]

**Comment by brainrepo** [comment]
Going to merge

**Pull request merged**